iac-common/src/test/java/org/sonar/iac/common/yaml/object/AttributeObjectTest.java

```diff
@@ -80,23 +81,30 @@ void reportIfValue() {
   }
 
   @Test
-  void isAbsentOrEmpty() {
+  void testReportIfAbsent() {
     TupleTree tree = parseTuple("a: b");
-    AttributeObject attr = AttributeObject.fromAbsent(ctx, "a");
-    boolean absentOrEmpty = attr.isAbsentOrEmpty(t -> true);
-    assertThat(absentOrEmpty).isTrue();
-
-    attr = AttributeObject.fromPresent(ctx, tree, "a");
-    absentOrEmpty = attr.isAbsentOrEmpty(t -> false);
-    assertThat(absentOrEmpty).isFalse();
-
-    attr = AttributeObject.fromPresent(ctx, tree, "a");
-    absentOrEmpty = attr.isAbsentOrEmpty(t -> true);
-    assertThat(absentOrEmpty).isTrue();
+    AttributeObject attr = AttributeObject.fromAbsent(ctx, "b");
+    attr.reportIfAbsent(tree.metadata(), "message2");
+    List<TestIssue> testIssues = raisedIssues.stream().filter(issue -> issue.message.equals("message2")).collect(Collectors.toList());
+    assertThat(testIssues).hasSize(1);
+    TestIssue testIssue = testIssues.get(0);
+    assertThat(testIssue.message).isEqualTo("message2");
+    assertThat(testIssue.secondaryLocations).isEmpty();
+    assertThat(testIssue.textRange).isEqualTo(tree.textRange());
+
+    attr = AttributeObject.fromPresent(ctx, tree, "b");
+    attr.reportIfAbsent(tree.metadata(), "message3");
+    testIssues = raisedIssues.stream().filter(issue -> issue.message.equals("message3")).collect(Collectors.toList());
+    assertThat(testIssues).isEmpty();
```

Comment:
This test method should be split into two as it tests two different behaviors.
 `reportIfAbsentShouldReportIssueOnAbsentObject` and `reportIfAbsentShouldNotReportIssueOnPresentObject`